packages/pigeon/lib/cpp_generator.dart

```diff
@@ -64,6 +70,26 @@ class CppOptions {
   }
 }
 
+/// Class that manages all Cpp header code generation.
+class CppGenerator extends Generator<CppOptions> {
+  /// Instantiates a Cpp Generator for the specified file type.
+  CppGenerator(this.fileType);
+
+  /// Specifies which file type (header or source) will be generated.
+  FileType fileType;
```

Comment:
Having this be a field still means we need to instantiate two generators. Couldn't this be a parameter to `generate` instead, so that a single generator can `generate` twice?

Reply:
It could be, but that would be added to all other generators as well. I played around with a few options that all were worse than this. If there's a piece of information I'm missing to make this work, I'd be happy to change it.

If it's worth adding an unused parameter onto all of the other generators, that's fine too. If that is what's best.

Reply:
Another option would be to change `generate` to take a map of filetype->sink. Then there's a single generation call, which gets all of the relevant sinks.

Reply:
That's not a bad idea. Should I eliminate the extra Generator adapters also? That's a somewhat larger process. If that's the endgoal though, I might as well make that happen now.